templates/workspaces/base/terraform/network/network.tf

```diff
@@ -75,6 +75,57 @@ resource "azurerm_subnet_route_table_association" "rt_services_subnet_associatio
   ]
 }
 
+data "azurerm_client_config" "current" {}
+
+resource "null_resource" "az_login_sp" {
+
+  count = var.arm_use_msi == true ? 0 : 1
+  provisioner "local-exec" {
+    command = "az login --service-principal --username ${var.arm_client_id} --password ${var.arm_client_secret} --tenant ${var.arm_tenant_id}"
+  }
+
+  triggers = {
+    timestamp = timestamp()
+  }
+
+}
+
+resource "null_resource" "az_login_msi" {
+
+  count = var.arm_use_msi == true ? 1 : 0
+  provisioner "local-exec" {
+    command = "az login --identity -u '${data.azurerm_client_config.current.client_id}'"
+  }
+
+  triggers = {
+    timestamp = timestamp()
+  }
+}
+
+resource "null_resource" "ws_core_peer_sync" {
```

Comment:
Why is it important to be done in TF and not via a porter step?

Reply:
See: https://github.com/hashicorp/terraform-provider-azurerm/issues/18444

Did it here as only wants to execute when the address spaces change.

Reply:
I see. Why did you choose the scripted way vs the _replace_triggered_by_ option?

Reply:
As this way avoids any downtime for existing services - the peer isn't disconnected.